src/posix/platform/trel_udp6.cpp

```diff
@@ -226,18 +235,29 @@ static void PrepareSocket(void)
 
     while (otPlatTimeGet() - startTime < TREL_SOCKET_BIND_MAX_WAIT_TIME_MSEC * USEC_PER_MSEC)
     {
-        if (bind(sSocket, (struct sockaddr *)&sockAddr, sizeof(sockAddr)) == -1)
-        {
-            VerifyOrDie(errno == EADDRNOTAVAIL, OT_EXIT_ERROR_ERRNO);
-        }
-        else
+        if (bind(sSocket, (struct sockaddr *)&sockAddr, sizeof(sockAddr)) != -1)
         {
             isSocketBound = true;
             break;
         }
+
+        if (errno == EADDRNOTAVAIL)
+        {
+            continue;
+        }
+
+        otLogCritPlat("[trel-plat] Failed to bind socket to %s (port %d) on TREL netif \"%s\"",
+                      Ip6AddrToString(&sInterfaceAddress), sUdpPort, sInterfaceName);
```

Comment:
```suggestion
        otLogCritPlat("[trel-plat] Failed to bind socket to %s (port %d) on TREL netif \"%s\": %s",
                      Ip6AddrToString(&sInterfaceAddress), sUdpPort, sInterfaceName, strerror(errno));
```

Reply:
`DieNow()` will add `errno` string. We can add it here as well but I think should be fine to have it in `DieNow()` log. Thoughts?

Reply:
Agree.